operators/pkg/controllers/keycloak_utils.go

```diff
@@ -0,0 +1,78 @@
+package controllers
+
+import (
+	"context"
+	"strings"
+
+	gocloak "github.com/Nerzal/gocloak/v7"
+	"k8s.io/klog"
+)
+
+// getClientID returns the ID of the target client given the human id, to be used with the gocloak library
+func getClientID(ctx context.Context, kcClient gocloak.GoCloak, token string, realmName string, targetClient string) (string, error) {
```

Comment:
To avoid this long list of common parameters, I would wrap the tuple (goclient, token, realm, client) in a struct and pass that to the functions.